lhotse/supervision.py

```diff
@@ -38,6 +38,17 @@ def trim(self, end: Seconds) -> 'SupervisionSegment':
         end_exceeds_by = max(0, self.end - end)
         return fastcopy(self, start=max(0, self.start), duration=self.duration - end_exceeds_by - start_exceeds_by)
 
+    def map(self, transform_fn: Callable[['SupervisionSegment'], None]) -> 'SupervisionSegment':
```

Comment:
Now we have an incosistent API for Supervisions and Cuts -> one accepts a function that returns nothing, and the other a function that returns a new object. Can you change this `transform_fn` to return a SupervisionSegment?

Reply:
What I thought is that `transform_fn` would be provided by users, so return `None` might be easier for users to write this function. :grin:
But I'm not sure about that. If needed, I can change that to return a `SupervisionSegment`.

Reply:
You're raising a good point. I think if we want to be super-user-friendly we could expose more specific functions like `SupervisionSet.transform_text(transform_fn: Callable[[str], str)`. I think `map` is more general-purpose and might confuse inexperienced users anyway, so no harm is done. But I also think it's good enough for now if it behaves consistently with the other map-like functions in Lhotse.

Also consider this: you cannot mutate a string or a tuple in Python, you would always have to return a new instance (like in the `transform_text` example). I'd like to pretend Lhotse's classes are immutable too, which makes it easier to create multiple "versions" of cuts/supervisions. Maybe it makes sense to make them truly immutable - I'll have to think more about this, I'm not super sure ATM.